src/foundry/client/ui/dialog.d.ts

```diff
@@ -1,287 +1,324 @@
+    ): Promise<Yes | No | null>;
 
     /**
-     * The buttons which are displayed as action choices for the dialog
+     * A helper factory method to display a basic "prompt" style Dialog with a single button
+     * @param config - Dialog configuration options
+     * @returns A promise which resolves when clicked, or rejects if closed
      */
-    buttons: Record<string, Button>;
+    static prompt<T>(
+      config: PromptConfig<T, JQuery> & { options?: { jQuery?: true }; rejectClose: false }
+    ): Promise<T | null>;
+    static prompt<T>(config: PromptConfig<T, JQuery> & { options?: { jQuery?: true } }): Promise<T>;
+    static prompt<T>(
+      config: PromptConfig<T, HTMLElement> & { options: { jQuery: false }; rejectClose: false }
+    ): Promise<T | null>;
+    static prompt<T>(config: PromptConfig<T, HTMLElement> & { options: { jQuery: false } }): Promise<T>;
+    static prompt<T>(config: PromptConfig<T, JQuery | HTMLElement> & { rejectClose: false }): Promise<T | null>;
+    static prompt<T>(config: PromptConfig<T, JQuery | HTMLElement>): Promise<T>;
 
     /**
-     * The name of the default button which should be triggered on Enter keypress
+     * Wrap the Dialog with an enclosing Promise which resolves or rejects when the client makes a choice.
+     * @param data - Data passed to the Dialog constructor
+     * @param options - Options passed to the Dialog constructor
+     * @param renderOptions - Options passed to the Dialog render call
+     * @returns A promise which resolves to the dialogs result.
      */
-    default?: string | undefined;
+    static wait<Options extends DialogOptions = DialogOptions>(
+      data: DialogData<JQuery>,
+      options?: Partial<DialogOptions> & { jQuery?: true },
+      renderOptions?: Application.RenderOptions<Options>
+    ): Promise<unknown>;
+    static wait<Options extends DialogOptions = DialogOptions>(
+      data: DialogData<HTMLElement>,
+      options?: Partial<DialogOptions> & { jQuery?: false },
+      renderOptions?: Application.RenderOptions<Options>
+    ): Promise<unknown>;
+    static wait<Options extends DialogOptions = DialogOptions>(
+      data: DialogData<JQuery | HTMLElement>,
+      options?: Partial<DialogOptions>,
+      renderOptions?: Application.RenderOptions<Options>
+    ): Promise<unknown>;
   }
-}
 
-/**
- * @typeParam Yes          - The value returned by the yes callback
- * @typeParam No           - The value returned by the no callback
- * @typeParam JQueryOrHtml - The parameter passed to the callbacks, either JQuery or HTMLElement
- */
-interface ConfirmConfig<Yes, No, JQueryOrHtml> {
   /**
-   * The confirmation window title
+   * @typeParam Yes          - The value returned by the yes callback
+   * @typeParam No           - The value returned by the no callback
+   * @typeParam JQueryOrHtml - The parameter passed to the callbacks, either JQuery or HTMLElement
    */
-  title?: string;
+  interface ConfirmConfig<Yes, No, JQueryOrHtml> {
```

Comment:
This should not be in the global namespace, as it is not defined as such by foundry. Our approach for things like these was as follows:

```typescript
declare global {
  namespace Dialog {
    interface PromptConfig {}
  }
}
```

You will also have to change references to it from just `PromptConfig` to `Dialog.PromptConfig`.